### PR TITLE
Improve error messaging for comparisons involving entrypoints

### PR DESCRIPTION
**Pull request by kiersten-stokes**
### What changes were proposed in this pull request?
This PR makes the error messaging more specific for cases where entrypoint names must match other names (e.g., catalog types `schema_name`s and runtime schema file names). 

This necessitates adding a logger to the `ElyraSchemasProvider`.

### How was this pull request tested?
This was manually tested to ensure that the correct errors are logged when no matching entrypoint name is encountered.

```
[E 2021-11-04 16:34:02.062 ElyraApp] No entrypoint with name 'local-directory-catalog' was found in group 'elyra.component.catalog_types' to match the 'schema_name' given in catalog 'Other Airflow Components'. Skipping...
```

```
[E 2021-11-04 16:34:01.571 ServerApp] No entrypoint with name 'kfp' was found in group 'elyra.pipeline.processor' to match the schema with the same name. Skipping...
```

 
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

File: elyra/metadata/schemasproviders.py
Comment (kevin-bates) [inline elyra/metadata/schemasproviders.py:117]: Nit: Would you mind keeping this as two statements?  The reason (I prefer this) is for debugging since, with two statements, the `return` statement serves as a good breakpoint to check things. I just feel this introduces a good "maintainability point" that otherwise would need to take place at the `get_schemas()` call point (which may vary) or in `get_local_schemas_by_schemaspace` which includes stepping across the various schemaspace ids.

**Review by kevin-bates — APPROVED** [review]
I'm approving now.  I'll let you decide if you want to consider my comment or not. I'm good either way (although I can't promise I won't make a similar comment down the road :smile: ).

**Review by ptitzler — APPROVED** [review]
LGTM/test results are as expected.

**Pull request merged**
